src/storage.rs

```diff
@@ -161,37 +192,38 @@ impl NdArray {
 }
 
 impl ReadChunk for NdArray {
-    fn read_chunk<R>(read: &mut R) -> Result<Self, Error>
+    fn read_chunk<R>(read: &mut R) -> Result<Self>
     where
         R: Read + Seek,
     {
-        let chunk_id = read.read_u32::<LittleEndian>()?;
-        let chunk_id = ChunkIdentifier::try_from(chunk_id)
-            .ok_or_else(|| format_err!("Unknown chunk identifier: {}", chunk_id))?;
-        ensure!(
-            chunk_id == ChunkIdentifier::NdArray,
-            "Cannot read chunk {:?} as NdArray",
-            chunk_id
-        );
+        ChunkIdentifier::check_read(read, ChunkIdentifier::NdArray)?;
```

Comment:
Similar comment like above, `check_read` is not self-explanatory for people either not very familiar with the fifu format or with the crate in general.

Reply:
I have changed all of these by `ensure_chunk_type` and `ensure_data_type` and added rustdoc to these methods.